charon/src/ast/types.rs

```diff
@@ -902,3 +896,37 @@ pub struct FunSig {
     pub inputs: Vec<Ty>,
     pub output: Ty,
 }
+
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize, Drive, DriveMut)]
+pub enum DynKind {
+    Dyn,
+    DynStar,
+}
```

Comment:
We will not support `*dyn` until the feature makes progress on the rustc side. Please remove this enum and support only `Dyn`.

Reply:
Well actually, `DynStar` just got removed from rustc so we should remove it also.